src/sparseconverter/__init__.py

```diff
@@ -377,67 +376,68 @@ def _build_converters(self):
+                    SCIPY_CSR_ARRAY, SCIPY_CSC_ARRAY):
+                for right in (SCIPY_COO, SCIPY_CSR, SCIPY_CSC, SCIPY_COO_ARRAY, SCIPY_CSR_ARRAY,
+                        SCIPY_CSC_ARRAY):
+                    self._complete((left, right), chain(_flatsig, _classes[right]))
             for left in SPARSE_COO, SPARSE_GCXS, SPARSE_DOK:
                 for right in NUMPY, CUDA:
-                    if (left, right) not in self._converters:
-                        self._converters[(left, right)] = _classes[left].todense
-            for left in SCIPY_COO, SCIPY_CSR, SCIPY_CSC:
+                    self._complete((left, right), _classes[left].todense)
+            for left in (SCIPY_COO, SCIPY_CSR, SCIPY_CSC, SCIPY_COO_ARRAY, SCIPY_CSR_ARRAY,
+                        SCIPY_CSC_ARRAY):
                 for right in NUMPY, CUDA:
-                    if (left, right) not in self._converters:
-                        self._converters[(left, right)] = _classes[left].toarray
-            for left in SCIPY_CSR, SCIPY_CSC:
+                    self._complete((left, right), _classes[left].toarray)
+            for left in SCIPY_CSR, SCIPY_CSC, SCIPY_CSR_ARRAY, SCIPY_CSC_ARRAY:
                 for right in SPARSE_COO, SPARSE_GCXS, SPARSE_DOK:
-                    if (left, right) not in self._converters:
-                        self._converters[(left, right)] = _convert_csc_csr_to_pydata_sparse(
-                            left, right
-                        )
+                    self._complete(
+                        (left, right),
+                        _convert_csc_csr_to_pydata_sparse(left, right)
+                    )
             for left in SCIPY_COO, :
                 for right in SPARSE_COO, SPARSE_GCXS, SPARSE_DOK:
-                    if (left, right) not in self._converters:
-                        self._converters[(left, right)] = _classes[right].from_scipy_sparse
-            self._converters[(SPARSE_COO, SCIPY_COO)] = chain(_flatsig, sparse.COO.to_scipy_sparse)
-            self._converters[(SPARSE_COO, SCIPY_CSR)] = chain(_flatsig, sparse.COO.tocsr)
-            self._converters[(SPARSE_COO, SCIPY_CSC)] = chain(_flatsig, sparse.COO.tocsc)
-            self._converters[(SPARSE_GCXS, SCIPY_COO)] = _GCXS_to_coo
-            self._converters[(SPARSE_GCXS, SCIPY_CSR)] = _GCXS_to_csr
-            self._converters[(SPARSE_GCXS, SCIPY_CSC)] = _GCXS_to_csc
+                    self._complete((left, right), _classes[right].from_scipy_sparse)
+
+            self._insert((SPARSE_COO, SCIPY_COO), chain(_flatsig, sparse.COO.to_scipy_sparse))
+            self._insert((SPARSE_COO, SCIPY_CSR), chain(_flatsig, sparse.COO.tocsr))
+            self._insert((SPARSE_COO, SCIPY_CSC), chain(_flatsig, sparse.COO.tocsc))
+            self._insert((SPARSE_GCXS, SCIPY_COO), _GCXS_to_coo)
+            self._insert((SPARSE_GCXS, SCIPY_CSR), _GCXS_to_csr)
+            self._insert((SPARSE_GCXS, SCIPY_CSC), _GCXS_to_csc)
```

Comment:
Here, we specialize `sparse <> SCIPY_{COO,CSR,CSC}` - do we need something similar for the `_ARRAY` variants? Is this done somewhere else?

Reply:
I've added a comment that explains this :-)